numpy/linalg/linalg.py

```diff
@@ -1997,8 +2001,6 @@ def lstsq(a, b, rcond="warn"):
     if rcond is None:
         rcond = finfo(t).eps * ldb
 
-    result_real_t = _realType(result_t)
-    real_t = _linalgRealType(t)
     bstar = zeros((ldb, n_rhs), t)
     bstar[:b.shape[0], :n_rhs] = b.copy()
```

Comment:
Not this PR, but when I looked at this before, I wondered what would be the point of `.copy()`; it is not like a view gets taken and this cannot be of much speed benefit for the whole routine.

Reply:
Yeah, the `copy` stuff here is weird.